### PR TITLE
Updates to Summary Sheet Checker & added BTL Processing

### PR DESCRIPTION
**Pull request by reedan88**
This PR includes updates to the Discrete Summary Sheet Checker as well as a notebook and some functions to generate the CTD Bottle File by parsing the individual .btl files.

### CODE REVIEW

File: discrete_summary_checker/example/Bottle_Processing.ipynb [hunk 1]
```diff
@@ -0,0 +1,1302 @@
...
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.9.12"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```
Comment (sbeaulieu) [inline discrete_summary_checker/example/Bottle_Processing.ipynb:1302]: OK (I used this notebook with Andrew to successfully compile Papa-09 .btl files.)

File: discrete_summary_checker/example/Bottle_Processing.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+Bottles.insert(0, "Cruise", "SKQ202208S")
+Bottles.head()
+
+# Save the results
+
+filename = "Papa-09_SKQ202208S_CTD_Bottle_Data_2022-11-16_ACR.xlsx"
+filepath = "/home/areed/Documents/OOI/reedan88/QAQC_Sandbox/Ship_data/data/Papa-09/" + filename
+
+Bottles.to_excel(filepath, index=False)
```
Comment (sbeaulieu) [inline discrete_summary_checker/example/Bottle_Processing.py:311]: OK (I used this with Andrew when compiling Papa-09 .btl files.)

File: discrete_summary_checker/example/Check_Format.ipynb [hunk 3]
```diff
@@ -988,7 +1163,7 @@
  ],
  "metadata": {
   "jupytext": {
-   "formats": "ipynb,auto:light"
+   "formats": "ipynb,py:light"
   },
   "kernelspec": {
```
Comment (sbeaulieu) [inline discrete_summary_checker/example/Check_Format.ipynb:1168]: OK (I ran this notebook for Pioneer-10)

File: discrete_summary_checker/example/Check_Format.py [hunk 4]
```diff
@@ -217,13 +217,13 @@ def is_same(series):
...
     Column('Discrete Salinity Replicate Flag', [MatchesPatternValidation(r"\*0|1{16}") | MatchesPatternValidation("-9999999")]),
     
     # Carbon System measurement: Check within ocean ranges; don't collect/measure pCO2
-    #     Alkalinity: Should be between 2100 - 2400
-    Column('Discrete Alkalinity [umol/kg]', [InRangeValidation(2100, 2400) | MatchesPatternValidation("-9999999")]),
+    #     Alkalinity: Should be between 2100 - 2450
+    Column('Discrete Alkalinity [umol/kg]', [InRangeValidation(2100, 2450) | MatchesPatternValidation("-9999999")]),
     Column('Discrete Alkalinity Flag', [MatchesPatternValidation(r"\*0|1{16}") | MatchesPatternValidation("-9999999")]),
     Column('Discrete Alkalinity Replicate Flag', [MatchesPatternValidation(r"\*0|1{16}") | MatchesPatternValidation("-9999999")]),
     #     DIC: Range should be 1900 - 2450
```
Comment (sbeaulieu) [inline discrete_summary_checker/example/Check_Format.py:229]: OK

File: discrete_summary_checker/data/CruiseInformation.csv [hunk 5]
```diff
@@ -244,3 +245,60 @@ SKQ202104S,Sikuliaq,2021-03-24T16:10:00,2021-04-07T18:00:00,Endurance 14 mooring
...
+TS20220714,Tenacious II,2022-07-14T02:00:00,2022-07-15T04:00:00,Pioneer 18c glider deployments
+EK20220715,Elakha,2022-07-15T15:13:00,2022-07-15T20:09:00,Endurance CE01ISSP CE02SHSP and glider ce_311 deployments
+PS20220720,Pacific Storm,2022-07-20T02:00:00,2022-07-21T23:40:00,Endurance CSPP and glider operations
+EK20220801,Elakha,2022-08-01T16:50:00,2022-08-01T20:45:00,Endurance CSPP and glider operations
+TN407,Thomas G. Thompson,2022-08-08T00:00:00,2022-09-16T23:59:59,Cabled (rvdata)
+TS20220822,Tenacious II,2022-08-22T08:00:00,2022-08-23T04:00:00,Pioneer 18d glider deployments
+TS20220911,Tenacious II,2022-09-11T08:00:00,2022-09-12T04:00:00,Pioneer glider cp_376 recovery
+WJ20220920,Warren Jr.,2022-09-20T09:00:00,2022-09-22T13:00:00,Pioneer 18e AUV deployment and test glider deployments
+TN408,Thomas G. Thompson,2022-09-21T19:05:00,2022-10-04T15:30:00,Endurance 17 moorings profilers and gliders
+TS20221006,Tenacious II,2022-10-06T10:00:00,2022-10-07T02:00:00,Pioneer glider cp_340 recovery
```
Comment (sbeaulieu) [inline discrete_summary_checker/data/CruiseInformation.csv:303]: OK (this appears to be a copy of the file in ooicgsn-asset-management/cruise)

**Review by sbeaulieu — APPROVED** [review]
I reviewed all 6 files. The seabird_ctd_name_map.xlsx file is a very helpful file used in Bottle_Processing

File: discrete_summary_checker/data/CruiseInformation.csv (hunk 5 above)
Comment (snwhite) [reply discrete_summary_checker/data/CruiseInformation.csv:303]: Concur it looks fine.  At least one more cruise has been added to the CruiseInformation file in OO/a-m

File: discrete_summary_checker/example/Check_Format.py (hunk 4 above)
Comment (snwhite) [reply discrete_summary_checker/example/Check_Format.py:229]: Looks fine.

File: discrete_summary_checker/example/Check_Format.ipynb (hunk 3 above)
Comment (snwhite) [reply discrete_summary_checker/example/Check_Format.ipynb:1168]: looks fine.

File: discrete_summary_checker/example/Bottle_Processing.ipynb (hunk 1 above)
Comment (snwhite) [reply discrete_summary_checker/example/Bottle_Processing.ipynb:1302]: Given that Stace was able to use it, I assume it's good.

File: discrete_summary_checker/example/Bottle_Processing.py (hunk 2 above)
Comment (snwhite) [reply discrete_summary_checker/example/Bottle_Processing.py:311]: Given that Stace was able to use it, I assume it's good.

**Review by snwhite — APPROVED** [review]

**Pull request merged**